prow/plugins/transfer-issue/transfer-issue.go

```diff
@@ -0,0 +1,96 @@
+package transferissue
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const pluginName = "transfer-issue"
+
+var (
+	transferRe = regexp.MustCompile(`(?mi)^/transfer(?:-issue)?\s*(.*)$`)
+)
+
+func init() {
+	plugins.RegisterGenericCommentHandler(pluginName, handleGenericComment, helpProvider)
+}
+
+func helpProvider(_ *plugins.Configuration, _ []config.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The transfer-issue plugin transfers a GitHub issue from one repo to another in the same organization.",
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/transfer-issue <destination repo in same org>",
+		Description: "Transfers an issue to a different repo in the same org.",
+		Featured:    true,
+		WhoCanUse:   "Org members.",
+		Examples:    []string{"/transfer-issue kubectl"},
+	})
+	return pluginHelp, nil
+}
+
+type githubClient interface {
+	GetRepo(org, name string) (github.FullRepo, error)
+	CreateComment(org, repo string, number int, comment string) error
+	IsCollaborator(org, repo, user string) (bool, error)
+	TransferIssue(org, dstRepoNodeID string, issueNodeID string) (*github.TransferIssueMutation, error)
+}
+
+func handleGenericComment(pc plugins.Agent, e github.GenericCommentEvent) error {
+	// TODO: check if this is an actual comment on an issue?
+	if e.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+	matches := transferRe.FindAllStringSubmatch(e.Body, -1)
+	if len(matches) == 0 {
+		return nil
+	}
+	if len(matches) != 1 && len(matches[0]) != 1 {
+		// TODO: post this as a comment and log
+		return fmt.Errorf("/transfer-issue must only be used once and with a single destination repo")
+	}
+	return handleTransfer(pc.GitHubClient, pc.Logger, e, matches[0][0])
+}
+
+func handleTransfer(gc githubClient, log *logrus.Entry, e github.GenericCommentEvent, dstRepoName string) error {
+	org := e.Repo.Owner.Login
+	srcRepo := e.Repo.Name
+	user := e.User.Login
+
+	dstRepo, err := gc.GetRepo(org, dstRepoName)
+	if err != nil {
+		log.WithError(err).Errorf("could not fetch destination repo: %s/%s", org, dstRepo)
+		// TODO: post comment. error here means something went wrong or that the repo doesn't exist (404)
+	}
+
+	isCollaboratorSrc, err := gc.IsCollaborator(org, srcRepo, user)
+	if err != nil {
+		log.WithError(err).Errorf("could not fetch if user: %s is collaborator of source repo: %s/%s", user, org, srcRepo)
+		// TODO: comment something went wrong?
```

Comment:
AFAIK there isn't any consensus about whether plugins shall respond with comments on internal-ish errors like this call. Most plugins don't comment on API errors like that.